base.go

```diff
@@ -293,8 +293,9 @@ func (j *jrequest) parseJSON(data []byte) error {
 	}
 
 	// Report an error for extraneous fields.
-	if len(extra) != 0 {
-		j.fail(code.InvalidRequest, "extra fields in request")
+	if l := len(extra); l != 0 {
+		j.fail(code.InvalidRequest, fmt.Sprintf("extra %d fields in request: %q",
+			l, extra))
```

Comment:
```suggestion
	if len(extra) != 0 {
		j.fail(code.InvalidRequest, fmt.Sprintf("extra fields in request: %q", extra))
```

I'm not convinced the count adds anything beyond listing the names.

That said—I wonder if we would be better off attaching this as ancillary error data, e.g., 

```go
j.err = DataErrorf(code.InvalidRequest, extra, "extra fields in request")
```

That would let the receiver recover the specific error names without parsing the error text.

Reply:
Both are valid points - I will update the PR with the suggestion.

Reply:
Done